app/src/main/java/activity/DetailsActivity.java

```diff
@@ -0,0 +1,163 @@
+    }
+
+    private class AsyncListStopsLoader extends AsyncTask<String, Void, List<Stop>> {
+
+        @Override
+        protected void onPostExecute(List<Stop> listStop) {
+            super.onPostExecute(listStop);
+            stopAdapter.setItemList(listStop);
+            stopAdapter.notifyDataSetChanged();
+            if(listStop == null || listStop.isEmpty()) {
+                Toast.makeText(getApplicationContext(), R.string.no_stops, Toast.LENGTH_LONG).show();
+            }
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+        }
+
+        @Override
+        protected List<Stop> doInBackground(String... params) {
+            List<Stop> listStop = new ArrayList<Stop>();
+
+            try {
+                RestServices restService = new RestServices();
+                String param =  "{\"params\":{\"routeId\": \"" + id + "\"}}";
+
+                JSONObject objJSON = restService.connectToRestService(params[0], param, getApplicationContext());
+                JSONArray arrJSON = objJSON.getJSONArray("rows");
+
+                for (int i = 0; i < arrJSON.length(); i++) {
+                    listStop.add(restService.convertStop(arrJSON.getJSONObject(i)));
+                }
+            }
+            catch(Exception ex) {
+                Toast.makeText(getApplicationContext(),ex.getMessage() , Toast.LENGTH_LONG).show();
+            }
+            return listStop;
+        }
+    }
+
+    private class AsyncListDeparturesLoader extends AsyncTask<String, Void, List<Departure>> {
+        private final ProgressDialog dialogDeparturesLoader = new ProgressDialog(DetailsActivity.this);
+
+        @Override
+        protected void onPostExecute(List<Departure> listDeparture) {
+            super.onPostExecute(listDeparture);
+            dialogDeparturesLoader.dismiss();
+
+            TextView textWeekDay = (TextView) findViewById(R.id.departure_weekday);
+            textWeekDay.setText(textWeekDay.getText() + strWeekDay);
+            TextView textSaturday = (TextView) findViewById(R.id.departure_saturday);
+            textSaturday.setText(textSaturday.getText() + strSaturday);
+            TextView textMonday = (TextView) findViewById(R.id.departure_sunday);
+            textMonday.setText(textMonday.getText() + strSunday);
+
+            if(listDeparture == null || listDeparture.isEmpty()) {
+                Toast.makeText(getApplicationContext(), R.string.no_departures, Toast.LENGTH_LONG).show();
+            }
+        }
+
+        @Override
+        protected void onPreExecute() {
+            super.onPreExecute();
+            dialogDeparturesLoader.setMessage(getString(R.string.loading_departures));
+            dialogDeparturesLoader.show();
+        }
+
+        @Override
+        protected List<Departure> doInBackground(String... params) {
+            List<Departure> listDeparture = new ArrayList<Departure>();
+
```

Comment:
There's some copy and paste in AsyncListStopsLoader and AsyncListDeparturesLoader. We must avoid copy and paste as much as we can.